apps/bancor-market-maker/contracts/BancorMarketMaker.sol

```diff
@@ -341,11 +405,12 @@ contract BancorMarketMaker is EtherTokenConstant, IsContract, AragonApp {
             return _msgValue == _value;
         }
 
-        return (
-            _msgValue == 0 &&
-            controller.balanceOf(_buyer, _collateral) >= _value &&
-            ERC20(_collateral).allowance(_buyer, address(this)) >= _value
-        );
+        bool buyerAllowanceAvailable = controller.balanceOf(_buyer, _collateral) >= _value
```

Comment:
Should we short circuit by adding the `! _noPreApproval ` condition first ?

Reply:
i think so